### PR TITLE
Added gomigrate for database versioning

### PR DESCRIPTION
**Pull request by Sri-Harsha06**

### CODE REVIEW

File: generator-jhipster-gomicro/generators/server/templates/gomicro/gomicro/migrate/migrate.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	"database/sql"
+	"fmt"
+	"strings"
+	app "<%= packageName %>/config"
+	"github.com/golang-migrate/migrate/v4"
+	_ "github.com/golang-migrate/migrate/v4/database/postgres"
+	_ "github.com/golang-migrate/migrate/v4/source/file"
+)
+
+func generateurl(databaseUrl, defaultDatabaseName, newDatabaseName string) string {
```
Comment (rohithteja-cmi) [inline generator-jhipster-gomicro/generators/server/templates/gomicro/gomicro/migrate/migrate.go]: generateUrl - case methods

File: generator-jhipster-gomicro/generators/server/templates/gomicro/gomicro/migrate/migration/000001_init_schema.down.sql [hunk 2]
```diff
@@ -0,0 +1 @@
+DROP TABLE IF EXISTS event;
```
Comment (craxkumar) [inline generator-jhipster-gomicro/generators/server/templates/gomicro/gomicro/migrate/migration/000001_init_schema.down.sql:1]: this is the first time the database will be created so there is no chance that the event table would exist from the starting 
right ?

File: generator-jhipster-gomicro/generators/server/templates/gomicro/gomicro/migrate/migrate.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+import (
+	"github.com/micro/micro/v3/service/logger"
+	_ "github.com/lib/pq"
+	"database/sql"
+	"fmt"
+	"strings"
+	app "<%= packageName %>/config"
+	"github.com/golang-migrate/migrate/v4"
+	_ "github.com/golang-migrate/migrate/v4/database/postgres"
```
Comment (craxkumar) [inline generator-jhipster-gomicro/generators/server/templates/gomicro/gomicro/migrate/migrate.go]: so this migrate.go file is for postgress specific so lets name it as postgress.go which will be under the migrate dir, so lets say in future we are have migrate file for the mongo then we will can have mongo.go under the migrate dir!
which will make the end user clear that those dir contains migrate files

**Review by craxkumar — CHANGES_REQUESTED** [review]

File: generator-jhipster-gomicro/generators/server/templates/gomicro/gomicro/migrate/migration/000001_init_schema.down.sql (hunk 2 above)
Comment (Sri-Harsha06) [reply generator-jhipster-gomicro/generators/server/templates/gomicro/gomicro/migrate/migration/000001_init_schema.down.sql:1]: No actually here, the "down" file is used to define the reverse actions that should be taken to undo the changes made by a specific migration's "up" file.So here in the case we want to roll back to previous version  the specific down file  here is to reverse what we have done in the up file by creating a table.

**Review by craxkumar — APPROVED** [review]
Review OK!

**Review by rohithteja-cmi — APPROVED** [review]

**Pull request merged**